e2e/tests/node_reboot.robot

```diff
@@ -11,22 +12,122 @@ Test Teardown    Cleanup test resources
+    Create statefulset 1 with rwx volume
+    Create statefulset 2 with rwo and strict-local volume
     FOR    ${i}    IN RANGE    ${LOOP_COUNT}
-        Delete replica on replica node to trigger replica rebuilding
-        During replica rebuilding, reboot replica node
-        Wait until replica on replica node rebuilt
-        Check data is intact
+        Keep writing data to deployment 0
+        Keep writing data to deployment 1
+        Keep writing data to deployment 2
+        Keep writing data to statefulset 0
+        Keep writing data to statefulset 1
+        Keep writing data to statefulset 2
+        Power off node 0 for 6 mins
+        Power off node 1 for 6 mins
+        Power off node 2 for 6 mins
+        Wait for longhorn ready
+        Check deployment 0 works
+        Check deployment 1 works
+        Check deployment 2 works
+        Check statefulset 0 works
+        Check statefulset 1 works
+        Check statefulset 2 works
+    END
+
+Reboot All Worker Nodes While Workload Heavy Writing
+    Create deployment 0 with rwo volume
+    Create deployment 1 with rwx volume
+    Create deployment 2 with rwo and strict-local volume
+    Create statefulset 0 with rwo volume
+    Create statefulset 1 with rwx volume
+    Create statefulset 2 with rwo and strict-local volume
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        Keep writing data to deployment 0
+        Keep writing data to deployment 1
+        Keep writing data to deployment 2
+        Keep writing data to statefulset 0
+        Keep writing data to statefulset 1
+        Keep writing data to statefulset 2
+        Restart all worker nodes
+        Wait for longhorn ready
+        Check deployment 0 works
+        Check deployment 1 works
+        Check deployment 2 works
+        Check statefulset 0 works
+        Check statefulset 1 works
+        Check statefulset 2 works
+    END
+
+Power Off All Worker Nodes For More Than Pod Eviction Timeout While Workload Heavy Writing
+    Create deployment 0 with rwo volume
+    Create deployment 1 with rwx volume
+    Create deployment 2 with rwo and strict-local volume
+    Create statefulset 0 with rwo volume
+    Create statefulset 1 with rwx volume
+    Create statefulset 2 with rwo and strict-local volume
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        Keep writing data to deployment 0
+        Keep writing data to deployment 1
+        Keep writing data to deployment 2
+        Keep writing data to statefulset 0
+        Keep writing data to statefulset 1
+        Keep writing data to statefulset 2
+        Power off all worker nodes for 6 mins
+        Wait for longhorn ready
+        Check deployment 0 works
+        Check deployment 1 works
+        Check deployment 2 works
+        Check statefulset 0 works
+        Check statefulset 1 works
+        Check statefulset 2 works
+    END
+
+Reboot Volume Node While Workload Heavy Writing
+    Create statefulset 0 with rwo volume
```

Comment:
We can do the same way you have done before by also adding deployment workload together.

Reply:
Created https://github.com/longhorn/longhorn/issues/6753 for this.